mlflow/models/evaluation/artifacts.py

```diff
@@ -21,3 +30,154 @@ def _save(self, output_artifact_path):
     def _load_content_from_file(self, local_artifact_path):
         self._content = pd.read_csv(local_artifact_path)
         return self._content
+
+
+class ParquetEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        self._content.to_parquet(output_artifact_path, compression="brotli")
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = pd.read_parquet(local_artifact_path)
+        return self._content
+
+
+class NumpyEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        np.save(output_artifact_path, self._content, allow_pickle=False)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = np.load(local_artifact_path, allow_pickle=False)
+        return self._content
+
+
+class JsonEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        with open(output_artifact_path, "w") as f:
+            json.dump(self._content, f)
+
+    def _load_content_from_file(self, local_artifact_path):
+        with open(local_artifact_path, "r") as f:
+            self._content = json.load(f)
+        return self._content
+
+
+class TextEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        with open(output_artifact_path, "w") as f:
+            f.write(self._content)
+
+    def _load_content_from_file(self, local_artifact_path):
+        with open(local_artifact_path, "r") as f:
+            self._content = f.read()
+        return self._content
+
+
+class PickleEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        with open(output_artifact_path, "wb") as f:
+            # Using protocol 4 because it is the highest supported protocol as of the
+            # introduction of PickleEvaluationArtifact. For more details regarding pickle
+            # protocols, refer to: https://docs.python.org/3/library/pickle.html#data-stream-format
+            pickle.dump(self._content, f, protocol=4)
```

Comment:
Instead of hardcoding the protocol, can we use the default protocol via `pickle.DEFAULT_PROTOCOL`? The default protocol is `4` for recent Python versions.

Reply:
I didn't do that because I thought it might be a good idea to maintain a fixed version to ensure future compatibility. i.e. in the event that DEFAULT_PROTOCOL changes to 5 in a future update, and that would cause MLflow unable to read pickle artifacts generated from prior versions.  Should I be concerned about this?

Reply:
It was 11 years of releases between minor version default assignment between Protocol 3 and 4. I think it's safe to use the default (also considering that a Py minor version is typically maintained for ~ 4 years, any model that's that old and hasn't retrained is likely a) not running in production any more or b) not providing any value to a company).

Reply:
Sounds good!